src/context/AppProvider/hooks/useAccountsFetchQuery.tsx

```diff
@@ -37,20 +37,26 @@ export const useAccountsFetchQuery = () => {
     return enabledWalletAccountIds.length > 0
   }, [isSnapInstalled, previousIsSnapInstalled, enabledWalletAccountIds.length])
 
+  // reset portfolio and tx history api states to ensure refetch on wallet switch
+  useEffect(() => {
+    if (!wallet) return
+
+    dispatch(portfolioApi.util.resetApiState())
+    dispatch(txHistoryApi.util.resetApiState())
+  }, [dispatch, wallet])
```

Comment:
q: This seems a bit rough but, do we have a world where wallet changes would rerun this for any sort of reasons?

I initially thought about something like wallet rename but it does not, so, seems sane so far, but if you have anything else in mind!

Reply:
Nothing else came to mind. It should be an actual hdwallet instance, so unless we are connecting to a new wallet, it should remain the same which would be the intention.